tests/hyph-except.sil

```diff
@@ -2,14 +2,20 @@
 \use[module=packages.retrograde,target=v0.14.17]
 \nofolios
 \language[main=tr]
-\lua{SILE.typesetter:typeset(SILE.showHyphenationPoints("rab'bin", "tr"))}
+\begin{lua}
+function typeset_hyphens (word, lang)
+   local hyphenator = SILE.typesetter.language.hyphenator
```

Comment:
Why should the hyphenator be accessed via the typesetter? Isn't this an inversion of control (vs. the typesetter invokes a hyphenator, which could a first class citizen)

Reply:
Vs. what not? In this layout the typesetter **does** invoke the language which then invokes it's own hyphenator. The typesetter is the most logical "close" thing for the document itself to reach for. There isn't a global for the currently active hyphenator, and we don't have any kind of sane state API (`SILE.documentState` being a disaster zone I'm chipping away at removing). Hence if the currently active typesetter (e.g. from here via `SILE.process()`) wants to find the current hyphenator for some reason, asking the typesetter for it's currently active language module and asking that language module what it's active hyphenator is seems like the only logical way.

Does that make sense or am I overlooking something?